processing/src/main/java/io/druid/segment/data/IndexedIntsWriter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.channels.WritableByteChannel;
+
+public interface IndexedIntsWriter extends Closeable
+{
+  public void open() throws IOException;
+
+  public void add(Object obj) throws IOException;
+
+  public long getSerializedSize();
+
+  public void writeToChannel(WritableByteChannel channel) throws IOException;
+}
```

Comment:
should this interface be called "ObjectsWriter" ? I don't see anything "indexed" or "int" specific in the interface.